static/js/read_receipts.js

```diff
@@ -0,0 +1,54 @@
+import $ from "jquery";
+import SimpleBar from "simplebar";
+
+import render_read_receipts from "../templates/read_receipts.hbs";
+import render_read_receipts_modal from "../templates/read_receipts_modal.hbs";
+
+import * as channel from "./channel";
+import {$t} from "./i18n";
+import * as loading from "./loading";
+import * as message_store from "./message_store";
+import * as overlays from "./overlays";
+import * as people from "./people";
+
+export function show_user_list(message_id) {
+    $("body").append(render_read_receipts_modal());
+    overlays.open_modal("read_receipts_modal", {
+        autoremove: true,
+        micromodal: true,
+        on_show: () => {
+            loading.make_indicator($("#read_receipts_modal .loading_indicator"));
+            channel.get({
+                url: `/json/messages/${message_id}/read_receipts`,
+                idempotent: true,
+                success(data) {
+                    const message_author_id = message_store.get(message_id).sender_id;
+                    const user_ids = data.user_ids.filter((id) => id !== message_author_id); // Do not show author.
+
+                    let users = user_ids.map((id) => people.get_by_user_id(id));
+                    users = users.filter((user) => !user.is_bot);
+                    users = users.map((user) => ({
+                        full_name: user.full_name,
+                        avatar_url: user.avatar_url,
+                    }));
```

Comment:
What's the thinking behind this compressed data structure?

We may in the future want to add support for clicking on a name in this widget open that user's basic profile popover to get more information on the user, so we're likely to want to include the `user_id`, but it seems like we could just skip this step entirely?

Reply:
Not sure how I ended up with this compressed data structure. Makes most sense to just skip this step. Removed this part of the code.